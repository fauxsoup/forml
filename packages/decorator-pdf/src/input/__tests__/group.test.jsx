import Group from "../group";
import Context from "@forml/context";
import React from "react";
import ReactPDF from "@react-pdf/renderer";
import { render } from "@testing-library/react";
import * as decorator from "../";

describe("renders", function() {
  let styles;
  let group;
  let form = { type: "fieldset", items: [] };

  beforeEach(function() {
    group = jest.fn(() => ({ border: "1px solid red" }));
    styles = {
      get group() {
        return group();
      },
    };
  });

  test("without formStyles", function() {
    const { container } = render(
      <ReactPDF.Page>
        <Context.Provider value={{ decorator }}>
          <Group form={form} />
        </Context.Provider>
      </ReactPDF.Page>
    );

    expect(container).toMatchSnapshot();
  });

  test("with relevant formStyles", function() {
    form = { type: "checkbox", styles, key: [] };

    const { container } = render(
      <ReactPDF.Page>
        <Context.Provider value={{ decorator }}>
          <Group form={form} />
        </Context.Provider>
      </ReactPDF.Page>
    );

    expect(container).toMatchSnapshot();
    expect(group).toHaveBeenCalled();
  });
});
