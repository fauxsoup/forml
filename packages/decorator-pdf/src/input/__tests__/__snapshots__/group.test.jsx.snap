// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with relevant formStyles 1`] = `
<div>
  <page>
    <view
      style="margin: 2pt; padding: 2pt; border: 1px solid red;"
    />
  </page>
</div>
`;

exports[`renders without formStyles 1`] = `
<div>
  <page>
    <view
      style="margin: 2pt; padding: 2pt;"
    />
  </page>
</div>
`;
